### PR TITLE
Refactor the SNTApplicationTest unit tests to function correctly

### PR DESCRIPTION
**Pull request by pmarkowsky**
The tests were originally written in a table style and were impacted by the lack of mocking the configurator. This caused issues with static rules to impact the unit tests.

Additionally added:
  *  Missing "goodbinary" for tests.
  *  Improved logging messages for critical binaries.

### CODE REVIEW

File: Source/santad/DataLayer/SNTRuleTable.m [hunk 1]
```diff
@@ -117,6 +117,8 @@ + (NSArray *)criticalSystemBinaryPaths {
       @"/Applications/Santa.app/Contents/MacOS/santasyncservice",
     ]];
 
+    // TODO for macOS 13 add the cryptexes prefix
```
Comment (russellhancox) [inline Source/santad/DataLayer/SNTRuleTable.m:120]: Add a link to a GH issue for this

File: Source/santad/SNTApplicationTest.m [hunk 2]
```diff
@@ -88,31 +104,105 @@ - (void)checkBinaryExecution:(NSString *)binaryName
   [mockES triggerHandler:msg.message];
 
   [self waitForExpectations:@[ expectation ] timeout:10.0];
+  NSString *clientModeStr;
+  if (clientMode == SNTClientModeLockdown) {
+    clientModeStr = @"LOCKDOWN";
+  } else {
+    clientModeStr = @"MONITOR";
+  }
```
Comment (russellhancox) [inline Source/santad/SNTApplicationTest.m:112]:
```objc
NSString *clientModeStr = (clientMode == SNTClientModeLockdown) ? @"LOCKDOWN" : @"MONITOR";

**Review by russellhancox — CHANGES_REQUESTED** [review]
Add `Source/santad/testdata/binaryrules/goodbinary` and `Source/santad/testdata/binaryrules/rules.db` to the allstar config

File: Source/santad/SNTApplicationTest.m (hunk 2 above)
Comment (pmarkowsky) [reply Source/santad/SNTApplicationTest.m]: Changed.

File: Source/santad/DataLayer/SNTRuleTable.m (hunk 1 above)
Comment (pmarkowsky) [reply Source/santad/DataLayer/SNTRuleTable.m]: I've created #886 but this was a left-over comment from other refactoring,  so I've removed it.

**Review by russellhancox — APPROVED** [review]

**Pull request merged**